### PR TITLE
[NPUW] Properties for npuw::CompiledModel are split to private and public

### PR DESCRIPTION
**Pull request by AsyaPronina**
### Details:
This PR defines public and private properties in `npuw::CompiledModel`:
 - All public properties and hints from `intel_npu::CompiledModel` become public properties in `npuw::CompiledModel`
 - All `*NPUW*` properties are private

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp [hunk 1]
```diff
@@ -627,15 +627,85 @@ void ov::npuw::CompiledModel::log_device_dist() const {
 void ov::npuw::CompiledModel::implement_properties() {
     // This function fills the map: {`property name`: `getter for property value`},
     // that can be used later to return requested properties by user.
-    // It does it in 4 steps:
+    // It does it in 3 steps:
     //
-    // 1. Create mappings for all NPUW-specific properties to getters of their
+    // 1. Create mappings for all coped from HETERO plugin properties.
```
Comment (TolyaTalamanov) [inline src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:632]: copied?

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp [hunk 2]
```diff
@@ -627,15 +627,85 @@ void ov::npuw::CompiledModel::log_device_dist() const {
...
-    // 3. Fill `m_all_supported_props` with all properties, mentioned above.
 
     // 1.
+    m_prop_to_opt = {
+        {ov::supported_properties.name(),
+         {ov::PropertyMutability::RO,
+          [&](const ::intel_npu::Config&) {
+              return m_all_supported_props;
+          }}},
+        {ov::device::properties.name(),
```
Comment (PatrikStepan) [inline src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:644]: Do we need this for NPUW? From what I know it was meant for proxy plugins, to provide different set of properties for each device: https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/openvino/runtime/properties.hpp#L905

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp (hunk 1 above)
Comment (sbutnari) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:632]: might be reordered ?

Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp]: Fixed, thanks!

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp (hunk 2 above)
Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp]: Thank you a lot, fixed!

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp [hunk 3]
```diff
@@ -654,15 +654,92 @@ void ov::npuw::CompiledModel::log_device_dist() const {
...
+
+#define GET_PLUGIN_PROP(property) return get_plugin()->get_property(property.name(), ov::AnyMap());
 
     // 1.
+    // OV Public
+    // =========
+    m_prop_to_opt = {{ov::supported_properties.name(),
+                      {ov::PropertyMutability::RO,
+                       [&](const ::intel_npu::Config&) {
+                           return m_all_supported_props;
```
Comment (smirnov-alexey) [inline src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:673]: What does this return btw?

Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:673]: Are private options hidden from this?

**Review by smirnov-alexey — APPROVED** [review]

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp (hunk 3 above)
Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:673]: Yes, true, private properties are defined by lack of them in that vector.

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp [hunk 4]
```diff
@@ -654,15 +654,92 @@ void ov::npuw::CompiledModel::log_device_dist() const {
...
+                           GET_PLUGIN_PROP(ov::hint::inference_precision);
+                       }}},
+                     {ov::hint::enable_cpu_pinning.name(),
+                      {ov::PropertyMutability::RO,
+                       [&](const ::intel_npu::Config&) {
+                           GET_PLUGIN_PROP(ov::hint::enable_cpu_pinning);
+                       }}},
+                     {ov::hint::model_priority.name(), {ov::PropertyMutability::RO, [&](const ::intel_npu::Config&) {
+                                                            GET_PLUGIN_PROP(ov::hint::model_priority);
+                                                        }}}};
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:734]: Alignment got broken here?

**Review by dmatveev — COMMENTED** [review]
It is hard to follow the change where previously-public properties become hidden. Please add more comments about it

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp (hunk 4 above)
Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:736]: Yes, but.. I don't know how to fix this.. This interesting format is proposed by clang-format fix..

**Review by dmatveev — APPROVED** [review]
